src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -74,6 +72,25 @@ class Crystal::Doc::Generator
       body = ""
     end
 
+    {raw_body, body}
+  end
+
+  def generate_docs_json(program_type, types)
+    raw_body, body = read_readme
+
+    json = Main.new(raw_body, Type.new(self, @program), repository_name)
+    puts json
+  end
+
+  def generate_docs_html(program_type, types)
+    copy_files
+    generate_types_docs types, @output_dir, types
+    generate_readme program_type, types
+  end
+
+  def generate_readme(program_type, types)
+    raw_body, body = read_readme
```

Comment:
The part for generating `body` is only needed in this method, so I suppose it should be moved here from `read_readme`.